### PR TITLE
[LM-10960] Added Custom Widget in Chat

### PR DESCRIPTION
**Pull request by ishaan1607**
## Description

**Features**
- Parsing of Custom Widget in Chat
- Sending Custom Widget as Attachment

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [X] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [X] Assignee added
- [X] Reviewer added
- [X] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: chatmm/src/main/java/com/likeminds/chatmm/conversation/view/adapter/databinder/ConversationWidgetItemViewDataBinder.kt [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                }
+
+                tvKey1.text = if (timeStamp != null) {
+                    "$timeStamp"
+                } else {
+                    "null timestamp"
+                }
+            } else {
+                clWidget.hide()
+            }
```
Comment (sidcr7-likeminds) [inline chatmm/src/main/java/com/likeminds/chatmm/conversation/view/adapter/databinder/ConversationWidgetItemViewDataBinder.kt:86]: add a simple comment to explain what is done here

File: chatmm/src/main/java/com/likeminds/chatmm/utils/ViewDataConverter.kt [hunk 2]
```diff
@@ -532,6 +536,18 @@ object ViewDataConverter {
             .build()
     }
 
+    private fun convertWidget(widget: Widget?): WidgetViewData? {
```
Comment (sidcr7-likeminds) [inline chatmm/src/main/java/com/likeminds/chatmm/utils/ViewDataConverter.kt:539]: add comment for the converter

**Review by sidcr7-likeminds — CHANGES_REQUESTED** [review]

File: chatmm/src/main/java/com/likeminds/chatmm/chatroom/detail/view/ChatroomDetailFragment.kt [hunk 3]
```diff
@@ -771,12 +781,65 @@ class ChatroomDetailFragment :
...
+            }
+
             clBottomBar.setOnClickListener {
                 initVisibilityOfAttachmentsBar(View.GONE)
             }
         }
     }
 
+    private fun onCustomWidgetAClicked() {
+        //todo add code
```
[the remark this answers is not in the document]
Comment (ishaan1607) [reply chatmm/src/main/java/com/likeminds/chatmm/chatroom/detail/view/ChatroomDetailFragment.kt:802]: Yes, for customer

File: chatmm/src/main/java/com/likeminds/chatmm/chatroom/detail/view/ChatroomDetailFragment.kt [hunk 4]
```diff
@@ -771,12 +781,65 @@ class ChatroomDetailFragment :
...
+            put("widget_type", widgetType)
+        }
+
+        postConversation(
+            metadata = metaData
+        )
+    }
+
+    private fun onCustomWidgetBClicked() {
+        //todo add code
```
[the remark this answers is not in the document]
Comment (ishaan1607) [reply chatmm/src/main/java/com/likeminds/chatmm/chatroom/detail/view/ChatroomDetailFragment.kt:823]: Yes, for customer

**Review by sidcr7-likeminds — APPROVED** [review]

**Pull request merged**